dot/telemetry/mailer.go

```diff
@@ -15,38 +15,45 @@ import (
 	"github.com/gorilla/websocket"
 )
 
-var (
-	// ErrInsufficientConnections occurs when BootstrapMailer is not able to connect to any endpoint
-	ErrInsufficientConnections = errors.New("insufficient connections")
-	ErrTimoutMessageSending    = errors.New("timeout sending telemetry message")
-)
-var messageQueue chan Message = make(chan Message, 256)
+//go:generate mockgen -destination=telemetry_mock.go -package $GOPACKAGE . Telemetry
+// Client interface holds the definitions to services send messages through telemetry
+type Client interface {
+	SendMessage(msg Message) error
+}
+
+var ErrTimoutMessageSending = errors.New("timeout sending telemetry message")
 
 type telemetryConnection struct {
 	wsconn    *websocket.Conn
 	verbosity int
 	sync.Mutex
 }
 
-// Handler struct for holding telemetry related things
-type mailer struct {
+// Mailer holds telemetry related attributes
+type Mailer struct {
 	messageQueue chan Message
 	connections  []*telemetryConnection
 	logger       log.LeveledLogger
+	enabled      bool
 }
 
-func newMailer(logger log.LeveledLogger) *mailer {
-	return &mailer{
-		messageQueue: messageQueue,
+func newMailer(enabled bool, logger log.LeveledLogger) *Mailer {
+	return &Mailer{
+		enabled:      enabled,
+		messageQueue: make(chan Message),
```

Comment:
Thinking about it will be better to have a mutex instead of a queue because the telemetry mailer is used by some services whose calls the `SendMessage` inside inner goroutines